### PR TITLE
[microsoft_dhcp] Add event.reason (long descriptions) to each event (DHCPv4)

### PR DESCRIPTION
**Pull request by nicpenning**
- Enhancement

## What does this PR do?

This will add the `Meaning` that is found in the DHCPv4 logs into `event.reason` as follows:

Event ID  Meaning
00	The log was started.
01	The log was stopped.
02	The log was temporarily paused due to low disk space.
10	A new IP address was leased to a client.
11	A lease was renewed by a client.
12	A lease was released by a client.
13	An IP address was found to be in use on the network.
14	A lease request could not be satisfied because the scope's address pool was exhausted.
15	A lease was denied.
16	A lease was deleted.
17	A lease was expired and DNS records for an expired leases have not been deleted.
18	A lease was expired and DNS records were deleted.
20	A BOOTP address was leased to a client.
21	A dynamic BOOTP address was leased to a client.
22	A BOOTP request could not be satisfied because the scope's address pool for BOOTP was exhausted.
23	A BOOTP IP address was deleted after checking to see it was not in use.
24	IP address cleanup operation has began.
25	IP address cleanup statistics.
30	DNS update request to the named DNS server.
31	DNS update failed.
32	DNS update successful.
33	Packet dropped due to NAP policy.
34	DNS update request failed.as the DNS update request queue limit exceeded.
35	DNS update request failed.
36	Packet dropped because the server is in failover standby role or the hash of the client ID does not match.
50+	Codes above 50 are used for Rogue Server Detection information.

The 50+ event code descriptions are found here: https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/dd183591(v=ws.10)#dhcp-server-logs-server-authorization-events

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by efd6** [comment]
For the commit message, can you provide a link for the messages that are below 50? Up-page of the link provided, there are some of them listed, but not the entirety. Do MS have a single canonical location for these details?

**Comment by nicpenning** [comment]
> For the commit message, can you provide a link for the messages that are below 50? Up-page of the link provided, there are some of them listed, but not the entirety. Do MS have a single canonical location for these details?

Interestingly enough, I could not find an extensive list from Microsoft. The full list lives in the eventog itself at the beginning of each log file when they get created each day.

Does that help?

**Comment by efd6** [comment]
Thanks, Nic. Yes that will do for the commit message. I'm surprised that Microsoft don't provide adequate documentation.

**Comment by nicpenning** [comment]
No problem. I tried to find it on their GitHub docs but these docs are fairly old and haven't made it to GH. If I ever do find it I will report accordingly.

**Review by efd6 — COMMENTED** [review]
Minor nit then LGTM

File: packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json [hunk 1]
```diff
@@ -702,4 +756,4 @@
             ]
         }
     ]
-}
\ No newline at end of file
+}
```
[the remark this answers is not in the document]
Comment (nicpenning) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:759]: Yes, I will see how that happens in the future.

**Comment by nicpenning** [comment]
Sounds good. I appreciate your thoroughness!

**Comment by nicpenning** [comment]
Anything else needed for this to be merged?

**Comment by nicpenning** [comment]
I fixed the merge conflicts. Can someone please review? Thanks!

File: packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json [hunk 2]
```diff
@@ -14,6 +14,9 @@
                 "kind": "event",
                 "original": "01,04/19/20,13:11:13,Stopped,,,",
                 "outcome": "success",
+                "reason": [
+                    "The log was stopped."
+                ],
```
Comment (andrewkroh) [inline packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:19]: [event.reason](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-reason) should not be an array (it's not specified as one in ECS).

Comment (nicpenning) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:19]: Agreed, thank you! Perhaps my pipeline is amiss. I need to see why the tests may have failed on this one.

Comment (nicpenning) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:19]: I don't think it is the fault of the pipeline. It also seems that I cannot convert an array to a string as a keyword type.

How do you recommend getting event.reason as a keyword?

File: packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml [hunk 3]
```diff
@@ -53,50 +53,66 @@ processors:
           action: log-start
           category:
             - process
+          reason:
+            - "The log was started."
```
Comment (andrewkroh) [inline packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml:57]:
```suggestion
          reason: "The log was started."
```

File: packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json (hunk 2 above)
Comment (nicpenning) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:19]: Silly me, I just need to remove the `-` on the new line. Fixing..

File: packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml (hunk 3 above)
Comment (nicpenning) [reply packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml]: :)

File: packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json (hunk 2 above)
Comment (nicpenning) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json]: Got it. Ran elastic-package build also and pushed.

**Comment by nicpenning** [comment]
Any last items on this? I think we are close.

**Comment by nicpenning** [comment]
👀

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**